### ISSUE TITLE
audioread.exceptions.NoBackendError comes up whenever i try to load my own data :(

### ISSUE THREAD

**Issue by Virus32wasmyname** [CorentinJ/Real-Time-Voice-Cloning#305]
Pls help!

Whenever I try to load in my own data set to clone from, it says: 

exception:
expected "str, byte..."

on the toolbox itself. When I try to load from my root with a saved WAV file (demo_toolbox.py -d <datasets_root>) it says: audioread.exceptions.NoBackendError on the terminal

**Comment by SugarpinkAI** [comment]
I am getting the same issue on a Windows 10 system.

**Comment by kraghavk** [comment]
I faced the same problem. There were two issues I had to resolve:
1. Looks like librosa does not like a Path object being sent to the load function. You will need to convert it to a string with full path, like this: str(path). You can change this in synthesizer/inference.py's load_preprocess_wav function.
2. Secondly, make sure your WAV file is 16000 Hz, 16 bit PCM. You can use audacity on windows to convert audio format.

Hope this helps.

Cheers
Raghu

**Comment by Simngscat** [comment]
win10 system
I solved this problem by installing ffmpeg and setting the environment variables
in Module audioread ffdec.py 
line 32 COMMANDS = ('ffmpeg', 'avconv')

**Comment by blue-fish** [comment]
If the above suggestions did not help you can try looking at librosa/librosa#219. Closing this issue as it does not strictly relate to the voice cloning toolbox.

**Comment by padmalcom** [comment]
I'd like to stress that this error also occurs when an audio file can not be read. Took me a day to figure out in my environment.

**Comment by blue-fish** [comment]
Thanks for reporting this @padmalcom . After #517 we check that librosa can load a sample audio on startup. If NoBackendError occurs while loading a file we can be fairly confident that the audio file is damaged. I'll consider adding a more helpful error message.

**Comment by blue-fish** [comment]
A pull request to add a helpful error message would be welcome if someone is willing to write it. Closing as this is a lower priority issue.

**Comment by god2106** [comment]
> I faced the same problem. There were two issues I had to resolve:
> 
> 1. Looks like librosa does not like a Path object being sent to the load function. You will need to convert it to a string with full path, like this: str(path). You can change this in synthesizer/inference.py's load_preprocess_wav function.
> 2. Secondly, make sure your WAV file is 16000 Hz, 16 bit PCM. You can use audacity on windows to convert audio format.
> 
> Hope this helps.
> 
> Cheers Raghu

def load_preprocess_wav(fpath):
        """
        Loads and preprocesses an audio file under the same conditions the audio files were used to
        train the synthesizer.
        """
        wav = librosa.load(str(fpath), hparams.sample_rate)[0]
        if hparams.rescale:
            wav = wav / np.abs(wav).max() * hparams.rescaling_max
        return wav
it is...but still not fixed?
